### PR TITLE
LibClassicInspector attempt to mute CanInspect error speech

### PR DESCRIPTION
**Pull request by ForestJ316**
From testing it seems like msg is always first and then sound after, so no race condition.

### CODE REVIEW

**Comment by perinazzoo** [comment]
its working? @ForestJ316

**Comment by ForestJ316** [comment]
From what I tested, yes

**Comment by perinazzoo** [comment]
@anzz1 any ETA to merge this?

**Comment by anzz1** [comment]
I can't test it right now, but I could merge it if you have tested it and there are no issues. @ForestJ316 @perinazzoo

**Comment by ForestJ316** [comment]
@anzz1 Been playing for a while and haven't noticed any issue so far. Wintergrasp seemed to be fine also, since I'm seeing a lot of ppl complaining about bgs triggering the error speech on curseforge comments. But I'm playing english locale. Doesn't seem like it should be an issue tho, unless blizzard decided to use different IDs for outofrange vocal errors on non-english locales?

**Comment by anzz1** [comment]
Very probably not, I would think constants such as that are indeed, constant. We'll see in the comments if that isn't the case then. :smile: 

What I'm more worried about is introducing taint, addon developers worst nemesis, which is the thing that causes those "Interface action failed because of an Addon" things which break the UI. Those kinds of pre-hooks can be dangerous since it taints the execution path that if an Blizzard function would call it and then do a "secure action", it could be blocked.

But such a thing like playing a error sound would probably happen at very tail-end of function execution and thus would not introduce taint, as was the case with the AddMessage hook.

Since you tested it and have had no problems, I can push it.

I'll get it merged this weekend and push an hotfix update. Thanks for your help !

File: Libs/LibClassicInspector/LibClassicInspector.lua [hunk 1]
```diff
@@ -2523,12 +2523,24 @@ oAddMessage = Detours:DetourHook(lib, UIErrorsFrame, "AddMessage", function(self
         skip_error = false
         local msg = ...
         if (msg == ERR_UNIT_NOT_FOUND or msg == ERR_INVALID_INSPECT_TARGET or msg == ERR_OUT_OF_RANGE) then
+            if GetCVar("Sound_EnableErrorSpeech") == "1" then
+				skip_error = true
```
Comment (anzz1) [inline Libs/LibClassicInspector/LibClassicInspector.lua:2527]:
```suggestion
                skip_error = true
```

File: Libs/LibClassicInspector/LibClassicInspector.lua [hunk 2]
```diff
@@ -2523,12 +2523,24 @@ oAddMessage = Detours:DetourHook(lib, UIErrorsFrame, "AddMessage", function(self
...
+            end
             return nil
         end
     end
     return oAddMessage(self, ...)
 end)
 
+local oPlayVocalErrorSoundID
+oPlayVocalErrorSoundID = Detours:DetourHook(lib, "PlayVocalErrorSoundID", function(vocalErrorSoundID, ...)
+	if vocalErrorSoundID == 10 and skip_error == true then
```
Comment (anzz1) [inline Libs/LibClassicInspector/LibClassicInspector.lua:2537]:
```suggestion
    if vocalErrorSoundID == 10 and skip_error == true then
```

File: Libs/LibClassicInspector/LibClassicInspector.lua [hunk 3]
```diff
@@ -2523,12 +2523,24 @@ oAddMessage = Detours:DetourHook(lib, UIErrorsFrame, "AddMessage", function(self
...
+		skip_error = false
```
Comment (anzz1) [inline Libs/LibClassicInspector/LibClassicInspector.lua:2538]:
```suggestion
        skip_error = false
```

File: Libs/LibClassicInspector/LibClassicInspector.lua [hunk 4]
```diff
@@ -2523,12 +2523,24 @@ oAddMessage = Detours:DetourHook(lib, UIErrorsFrame, "AddMessage", function(self
...
+		return nil
```
Comment (anzz1) [inline Libs/LibClassicInspector/LibClassicInspector.lua:2539]:
```suggestion
        return nil
```

File: Libs/LibClassicInspector/LibClassicInspector.lua [hunk 5]
```diff
@@ -2523,12 +2523,24 @@ oAddMessage = Detours:DetourHook(lib, UIErrorsFrame, "AddMessage", function(self
...
+	end
```
Comment (anzz1) [inline Libs/LibClassicInspector/LibClassicInspector.lua:2540]:
```suggestion
    end
```

File: Libs/LibClassicInspector/LibClassicInspector.lua [hunk 6]
```diff
@@ -2523,12 +2523,24 @@ oAddMessage = Detours:DetourHook(lib, UIErrorsFrame, "AddMessage", function(self
...
+	return oPlayVocalErrorSoundID(vocalErrorSoundID, ...)
```
Comment (anzz1) [inline Libs/LibClassicInspector/LibClassicInspector.lua:2541]:
```suggestion
    return oPlayVocalErrorSoundID(vocalErrorSoundID, ...)
```

**Review by anzz1 — CHANGES_REQUESTED** [review]
tabs to spaces

**Comment by anzz1** [comment]
Merged, released in 0.2.8 and pushed to addon channels. Thanks !

**Pull request merged**